### PR TITLE
CH-185: Support bundle screen in Maintenance section

### PR DESCRIPTION
**Pull request by NareshGopal**
# Problem Statement
- Build the support bundle screen in the maintenance section

# Design
-  Build the support bundle page as per the [design](https://www.figma.com/file/OGlWdQIRAntRCuROdQhzJB/Maintenance-Landing-Screen?node-id=2%3A82)

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: src/components/maintenance/LrMaintenanceSupportBundle.vue [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    return isDisable;
+  }
+
+  public dateRange = {
+    startDate: 0,
+    endDate: 0,
+  };
+
+  @Watch("timePeriod")
+  public printDates() {
```
Comment (sujit-h-singh) [inline src/components/maintenance/LrMaintenanceSupportBundle.vue:163]: what is this function used for? to compare?

File: tests/DummyData.json [hunk 2]
```diff
@@ -423,5 +423,48 @@
...
+        "status": "In progress",
+        "parameters": {
+          "component": "CSM",
+          "node": "Node1"
+        }
+      }
+    ]
+  },
+  "maintenance/node-list": {
+    "data": ["Node1", "Node2", "Node3", "Node4"]
```
Comment (sujit-h-singh) [inline tests/DummyData.json:468]: Most probable this might be array of objects.

**Review by sujit-h-singh — CHANGES_REQUESTED** [review]

File: src/components/maintenance/LrMaintenanceSupportBundle.vue [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+                  <v-icon
+                    color="primary"
+                    @click="timePeriodMenu = !timePeriodMenu"
+                    >mdi-calendar-month-outline</v-icon
+                  >
+                </template>
+              </v-text-field>
+            </template>
+            <v-date-picker
+              color="csmprimary"
```
[the remark this answers is not in the document]
Comment (NareshGopal) [reply src/components/maintenance/LrMaintenanceSupportBundle.vue:67]: If not mentioned, the highlight color of the dates are blue.

File: src/components/maintenance/LrMaintenanceSupportBundle.vue [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+    }
+    return isDisable;
+  }
+
+  public dateRange = {
+    startDate: 0,
+    endDate: 0,
+  };
+
+  @Watch("timePeriod")
```
[the remark this answers is not in the document]
Comment (NareshGopal) [reply src/components/maintenance/LrMaintenanceSupportBundle.vue]: Modified as per the feedback.

File: src/components/maintenance/LrMaintenanceSupportBundle.vue (hunk 1 above)
Comment (NareshGopal) [reply src/components/maintenance/LrMaintenanceSupportBundle.vue:163]: When sending dates to the backend we need to send them in UNIX format. The date picker component stores date values in string format. So before sending the data, we need to make this conversion.

File: tests/DummyData.json (hunk 2 above)
Comment (NareshGopal) [reply tests/DummyData.json]: Modified as per the feedback.

File: src/components/maintenance/LrMaintenanceSupportBundle.vue (hunk 1 above)
Comment (NareshGopal) [reply src/components/maintenance/LrMaintenanceSupportBundle.vue:163]: Unix time conversion will be done only when making the API call.

**Review by sujit-h-singh — APPROVED** [review]

**Pull request merged**